src/name_core.jl

```diff
@@ -41,19 +41,25 @@ e.g `dim((:a, :b)) == (a=1, b=2)`.
 If the second `name` argument is given, them the dimension corresponding to that `name`,
 is returned.
 e.g. `dim((:a, :b), :b) == 2`
-If that `name` is not found then `0` is returned.
+If that `name` is not found then an error is thrown.
 """
-function dim(dimnames::Tuple)
+function dim(dimnames::Tuple)  # TODO: Remove this ?
```

Comment:
```suggestion
function dims(dimnames::Tuple)
```
Is this name already taken? I think this is a useful function but it is unlike `dim`.

Reply:
It is not taken.
We just can't use it easily it things that have the kwarg `dims`.

But I do not think tthe `dims` function is that useful.
Not in its current form
Particularly since it breaks if you have multiple wildcard dimensions.
Which everything else in the package supports now as of this PR.
(except things that take the dimensions as kwargs like `getindex` but that is reasonably not to support since it has no canonical answer).

Reply:
Functions that have that kwarg would just have to qualify it in that scope. Not a big deal; we wouldn't get conflict warnings or anything.

I'm not really happy with the wildcard stuff. But for now you're the only user and it helps you so better to have it than not.

Reply:
I didn't like the wildcard stuff and first, but I am convinced it is essential to allow painless interop with other array types and with operations like transpose.

Though we might get rid of transpose at least from common usage if we add automatic dimension permuta tion. And we might even be able to cansually force all partially named NamedDimsArrays to only have their wildcards at the back.
Need to open an issue to discuss this though

Reply:
> Functions that have that kwarg would just have to qualify it in that scope. Not a big deal; we wouldn't get conflict warnings or anything.

Correct.
But I still don't think the function that returns a NamedTuple is that useful

Reply:
> But I still don't think the function that returns a NamedTuple is that useful

Yeah okay. Part of me will die though. RIP NamedTuples for everything

Maybe you'll be able to convince me that we need a MultiNamedTuple

Reply:
I had a lot of fun using NamedTuple Tricks.
But alas this PR removed all of them

Reply:
Stupid features getting in the way of elegance